### DISCUSSION TITLE
Change shortcut to open Palette

### DISCUSSION THREAD

**Discussion by transcental** [community/community#24058] [Other Feature Feedback, Questions, & Ideas]
<p>To make it more consistent with command palettes in other websites/applications, maybe you should consider changing the key bind to ⌘+⇧+P?</p>
<p>This is the same as Visual Studio Code and numerous other IDEs and text editors.</p>

**Comment by lol768** [comment]
<p>Yes, <em>please</em> don’t hijack my browser’s shortcut to focus the search bar!</p>

**Comment by JosephTLyons** [comment]
<p>If they would just allow the key binding to be customized, then people could do what they want.  I also prefer the idea of Command + Shift + P, to be consistent with all the editors that use this convention.</p>
<aside class="quote" data-post="1" data-topic="208859">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/custom-key-binding/208859">Custom Key Binding</a> <a class="badge-wrapper  bullet" href="/c/command-palette/71"><span class="badge-category-bg" style="background-color: #E45735;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="This is the space to discuss what we have named Command Palette! Submit feedback, and share your experiences, here!">Command Palette</span></a>
  </div>
  <blockquote>
    The new command palette action at <a href="http://GitHub.com" rel="noopener nofollow ugc">GitHub.com</a> is fantastic.  Is it possible to allow the user to make a custom keybinding to it?  Most editors use Command + Shift + P to launch the command palette.
  </blockquote>
</aside>

**Comment by anaisbetts** [comment]
<p>The entire world uses Cmd/Ctrl-Shift-P, please don’t reinvent this!</p>

**Comment by inkblotty** [comment]
<p>GH is exploring key remappings so these shortcuts can be fully customized in the future.</p>
<aside class="quote quote-modified" data-post="3" data-topic="208859">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/custom-key-binding/208859/3">Custom Key Binding</a> <a class="badge-wrapper  bullet" href="/c/command-palette/71"><span class="badge-category-bg" style="background-color: #E45735;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="This is the space to discuss what we have named Command Palette! Submit feedback, and share your experiences, here!">Command Palette</span></a>
  </div>
  <blockquote>
    (edited original comment to be more accurate) 
Thanks <a class="mention" href="/josephtlyons">@josephtlyons</a>, <a class="mention" href="/dorner">@dorner</a> &amp; <a class="mention" href="/thatgeoguy">@thatgeoguy</a> for the feedback! We actually originally used command+p for our internal staff ship but couldn’t get around bugs with it inconsistently pulling up the print dialog. Therefore we chose to go with command+k (and Windows/Linux equivalents) with the command+option+k alternate. I’ve shared this with our team to discuss a workaround. For now if it’s breaking your browser experience you are able to turn off the f…
  </blockquote>
</aside>

**Comment by gaknoll** [comment, marked as answer]
<p>As <a class="mention" href="/inkblotty">@inkblotty</a> reposted, we did in fact start with command+p and command+shift+p but were unable to consistently override the browser print dialog (which was quite frustrating). We have gotten a lot of feedback on this already and will be discussing as a team how we could improve this experience and/or give users more control / customization.</p>
<p>In the meantime, if overriding the browser search bar command is too frustrating, you can disable the command palette via Feature Preview in your profile dropdown while we work on a solution for you.</p>

**Comment by dtinth** [comment]
<p>Here’s a userland workaround: This userscript remaps Cmd+Shift+P to Cmd+K. Works on Mac only. Maybe someone can create a version of this that also works on Windows/Linux.</p>
<pre><code class="lang-auto">// ==UserScript==
// @name         Remap GitHub Command Palette to Cmd+Shift+P
// @namespace    http://dt.in.th/
// @version      0.1
// @description  Remap GitHub Command Palette to Cmd+Shift+P. Only works on Mac.
// @author       dtinth
// @match        https://github.com/*
// @icon         https://www.google.com/s2/favicons?domain=github.com
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    document.addEventListener('keydown', (e) =&gt; {
        if (e.key.toLowerCase() === 'p' &amp;&amp; e.shiftKey &amp;&amp; e.metaKey) {
            e.stopPropagation()
            e.preventDefault()
            document.body.dispatchEvent(new KeyboardEvent('keydown', { key: 'k', metaKey: true, bubbles: true }));
        }
    }, { capture: true })
})();
</code></pre>

**Comment by gaknoll** [comment]
<p>Hi everyone! We just shipped a <a href="https://github.blog/changelog/2022-02-03-customizing-command-palette-keyboard-shortcuts-beta/">new feature</a> that allows you to customize the keyboard shortcuts used to open the command palette. You can now customize the keyboard shortcuts for opening the command palette in search and command modes in the <a href="https://github.com/settings/accessibility">Accessibility section</a> of your user settings.</p>

**Comment by superjamie** [comment]
Ctrl+Shift+P is Private Browsing in Firefox. Don't hijack this.
